packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -229,6 +230,39 @@ export interface CanaryProps {
    * @default - no rules applied to the generated bucket.
    */
   readonly artifactsBucketLifecycleRules?: Array<s3.LifecycleRule>;
+
+  /**
+   * Canary Artifacts in S3 encryption mode.
+   * Artifact encryption is only supported for canaries that use Synthetics runtime
+   * version `syn-nodejs-puppeteer-3.3` or later.
+   *
+   * @default - Artifacts are encrypted at rest using an AWS managed key
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_artifact_encryption.html
+   */
+  readonly artifactS3EncryptionMode?: ArtifactsEncryptionMode;
+
+  /**
+   * The KMS key to be used to encrypt the data.
+   *
+   * @default - no kms key if mode = S3_MANAGED. A key will be created if one is not provided and mode = KMS.
+   */
+  readonly kmsKey?: kms.IKey;
```

Comment:
```suggestion
   * The KMS key used to encrypt canary artifacts.
   *
   * @default - no kms key if `artifactS3EncryptionMode` is set to `S3_MANAGED`. A key will be created if one is not provided and `artifactS3EncryptionMode` is set to `KMS`.
   */
  readonly artifactS3KmsKey?: kms.IKey;
```
What about a more meaningful variable name?